srunner/scenarios/parking_exit.py

```diff
@@ -0,0 +1,174 @@
+    """
+
+    def __init__(self, world, ego_vehicles, config, debug_mode=False, criteria_enable=True,
+                 timeout=180):
+        """
+        Setup all relevant parameters and create scenario
+        and instantiate scenario manager
+        """
+        self._world = world
+        self._map = CarlaDataProvider.get_map()
+        self.timeout = timeout
+        self._parking_point = convert_dict_to_location(
+            config.other_parameters['parking_point'])
+
+        if 'front_vehicle_distance' in config.other_parameters:
+            self._front_vehicle_distance = float(
+                config.other_parameters['front_vehicle_distance']['value'])
+        else:
+            self._front_vehicle_distance = 20  # m
+
+        if 'behind_vehicle_distance' in config.other_parameters:
+            self._behind_vehicle_distance = float(
+                config.other_parameters['behind_vehicle_distance']['value'])
+        else:
+            self._behind_vehicle_distance = 5  # m
+
+        if 'end_distance' in config.other_parameters:
+            self._end_distance = float(
+                config.other_parameters['end_distance']['value'])
+        else:
+            self._end_distance = 25  # m
+
+        super(ParkingExit, self).__init__("ParkingExit",
+                                          ego_vehicles,
+                                          config,
+                                          world,
+                                          debug_mode,
+                                          criteria_enable=criteria_enable)
+
+    def _create_behavior(self):
+        """
+        Deactivate OutsideRouteLanesTest, then move ego to the parking point, generate blocking vehicles in front of and behind the ego.
+        After ego drives away, actviate OutsideRouteLanesTest, end scenario.
+        """
+
+        sequence = py_trees.composites.Sequence()
+
+        parking_transform = front_transform = carla.Transform()
+        parking_transform.location = self._parking_point
+        # To get rotation like this because the return value of the
+        # get_waypoint(self, location, project_to_road=True, lane_type=carla.LaneType.Parking)
+        # would be weird in some maps. So using get_waypoint(self._parking_point) to get a waypoint in driving lane
+        # which has the same rotation with the parking point is a safe choice.
+        parking_transform.rotation = self._map.get_waypoint(
+            self._parking_point).transform.rotation
+
+        # Put blocking vehicles
+        front_transform = get_same_dir_transform_by_distance(
+            parking_transform, self._front_vehicle_distance)
+        front_transform.location.z += 0.5
+        behind_transform = get_same_dir_transform_by_distance(
+            parking_transform, -1*self._behind_vehicle_distance)
+        behind_transform.location.z += 0.5
+
+        actor_source_front = ActorSource(
```

Comment:
The actor creation can be done when initializing the scenario. In that way, we can use the CarlaDataProvider.request_new_actor(), instead of an atomic behavior.

Also, ActorSource isn't really meant to be used like this

Reply:
Using CarlaDataProvider.request_new_actor() instead of ActorSource